ethcore/src/spec/spec.rs

```diff
@@ -785,7 +785,7 @@ impl Spec {
 			let tx = Transaction {
 				nonce: self.engine.account_start_nonce(0),
 				action: Action::Call(a),
-				gas: U256::from(50_000_000), // TODO: share with client.
+				gas: U256::default(),
```

Comment:
doesn't seem right to provide no gas. then no work will be done and we will almost certainly end up in the wrong state. 50M is an arbitrary amount but we could also set to `U256::max_value` -- the key thing is really that the `env_info`'s gas limit is at least the amount of gas provided to the transaction, which seems to be the failure here.

Reply:
I think for `genesis_epoch_data`, the goal is to get an initial list of validators. when the `ValidatorSet` is a contract, we have to call a function on that contract to extract the validators. This function needs some gas to read out the validators from storage, allocated memory for them, and ABI-encode the return data.

The thought process I've been following so far with ValidatorSet contracts is that supplying infinite gas is fine because the contract is actually part of the consensus process and shouldn't be subject to the metering rules or have an escape-hatch from turing-completeness. The same should apply here.